app/sample_manifest_excel/sample_manifest_excel/upload/processor/base.rb

```diff
@@ -149,7 +149,7 @@ def check_for_barcodes_unique
         # Return the row of the first encountered barcode mismatch
         # rubocop:todo Metrics/MethodLength, Metrics/AbcSize
         def duplicate_barcodes # rubocop:todo Metrics/CyclomaticComplexity
-          return unless upload.respond_to?('rows')
+          return unless upload.respond_to?(:rows)
```

Comment:
Huh, I know this was an auto-correct, and I'm definitely one for using symbols in this context, but this check is a little weird, as we only seem to have a single upload class, that always responds to rows. So no idea why we are performing this check. Perhaps we're creating a processor without an upload in the tests, but none of the other methods bother performing this check.

Not something we should be touching here, especially as it probably *was* added for a reason.

Reply:
So I think this is a standard validation guard to prevent cascading validations being triggered. Agreed the upload should always have rows but if there is something wrong with the manifest and there are no rows this would trigger a 500 error which would obfuscate the validation errors.

Reply:
But in this case upload would still respond to rows, it would just return nil (or ideally an empty array). So this guard would catch cases where upload itself was nil. Maybe this happens if a file isn't provided?